doc/en/api/1.0.0/authenticationproviders.yaml

```diff
@@ -0,0 +1,332 @@
+swagger: '2.0'
+info:
+  title: Authentication-provider REST API
+  version: 1.0.0
+  description: |
+    Manage, order, enable and disable GeoServer authentication providers.
+
+    ### Representation notes
+    - **XML:** each provider is serialized with its concrete class name as the element name
+      (e.g. `<org.geoserver.security.config.UsernamePasswordAuthenticationProviderConfig>`).
+      For POST/PUT send the provider element **directly** (no `<authProvider>` wrapper).
+    - **JSON:** requests may be **either** a bare object `{ ... }` **or** an envelope:
+        - Single: `{ "authProvider": { ... } }`
+        - List:   `{ "authProviders": [ { ... }, ... ] }`
+      (The controller also accepts a single-item `"authProviders"` array.)
+    - **Create rules:** `className` is required on POST.
+    - **Update rules:** path `{providerName}` **must** equal payload `name`. `className` cannot change
+      (you may omit it in the payload to keep the existing value).
+    - **Enable/disable semantics:** the active order in `<authProviderNames>` determines which providers
+      are enabled. Presence = enabled; absence = disabled.
+
+host: localhost:8080
+basePath: /geoserver/rest
+schemes: [http]
+security:
+  - basicAuth: []
+
+paths:
+  /security/authProviders:
```

Comment:
As a general practice is it best to try and keep uri's lowercase, so this would become `/security/authproviders`